backend/src/test/java/com/bikes/backend/service/BikeServiceTest.java

```diff
@@ -3,43 +3,120 @@
 import com.bikes.backend.model.Bike;
 import com.bikes.backend.repository.BikeRepository;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
-import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
+import java.util.Optional;
 
-import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.*;
+
 class BikeServiceTest {
 
-    BikeRepository bikeRepository = mock(BikeRepository.class);
-    BikeService bikeService = new BikeService(bikeRepository);
+	BikeRepository bikeRepository = mock(BikeRepository.class);
+	IdService idService = mock(IdService.class);
+    BikeService bikeService = new BikeService(bikeRepository, idService);
 
-    Bike testBike = new Bike("testBike", "testId");
-    List<Bike> expectedBikes = List.of(testBike);
-    List<Bike> expectedBikesEmpty = new ArrayList<>();
+	Bike testBike = new Bike("testId", "testBike");
+	List<Bike> expectedBikes = List.of(testBike);
```

Comment:
Watch out: List.of() creates an immutable List . Keep it in mind while testing...

Reply:
Ok